src/components/editor/spec.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import test from 'tape';
+import { shallow, mount } from 'enzyme';
+
+import { ContentState, ContentBlock, CharacterMetadata, convertToRaw, genKey } from 'draft-js';
+import { fromJS, List, Repeat } from 'immutable';
+
+import EditorFactory from './';
+import UpstreamEditor from 'draft-js-plugins-editor';
+
+const genContent = () => {
+  const text = 'Write here';
+  const contentState = ContentState.createFromBlockArray([
+    new ContentBlock({
+      key: genKey(),
+      type: 'unstyled',
+      text,
+      characterList: List( Repeat( CharacterMetadata.EMPTY, text.length ) )
+    }),
+  ]);
+  return convertToRaw(contentState);
+};
+
+const getState = (el) => {
+  return el.find('span').text();
+};
+
+test('Editor', t => {
+  const Editor = EditorFactory(React);
+  t.plan(4);
+
+  const content = genContent();
+  const instance = shallow(<Editor content={content} />);
+
+  t.equals(getState(instance), 'Reading');
+  t.equals(instance.find(UpstreamEditor).length, 1);
+
+  instance.simulate('click');
+
+  t.equals(getState(instance), 'Writing');
```

Comment:
I'm not sure to which other comment you're referring.

BUT. Again, I want to draw the distinction between a unit test and an integration test. Typing is handled by Draft, so as far as a unit test is concerned, its functionality is in the trust. All we care about for *this* component's unit test is *this* component's functionality. For example, on click, does the editor component get a changed `readOnly` only prop? And we are supposed to trust that `onBlur` works as defined, so we don't care for a *unit test* whether it does - we only care if the callback we pass to that Editor prop does what it's supposed to.

Similarly with pasting. The plugin is a unit, so it's tested separately (see another comment in this review). This component is another unit. So for this component's test, we only need ensure we are passing the plugin downstream - we assume that it works as defined, so we don't test its functionality again here.

But in an *integration test*, we test that it all works together. We'll be working to add those as "end-to-end" tests (they'll be separate github issues) once we start incorporating these components into routes. In the meantime, we can see if these isolated components perform as expected in integration through storybook stories.